src/content/docs/en/reference/api-reference.mdx

```diff
@@ -2137,6 +2137,355 @@ export const onRequest = defineMiddleware(async (context, next) => {
+
+### `supportsViewTransitions`
+
+<p>
+
+**Type:** `boolean`<br />
+<Since v="3.2.0" />
+</p>
+
+A constant that indicates whether View Transitions are supported and enabled in the current browser.
+
+### `transitionEnabledOnThisPage`
+
+<p>
+
+**Type:** `boolean`<br />
+<Since v="3.2.0" />
+</p>
+
+Whether the current page has the View Transitions enabled for client-side navigation. This can be used to make reusable components that adapt to when they are used on pages with and without View Transitions.
+
+### `getFallback`
+
+<p>
+
+**Type:** `() => 'none' | 'animate' | 'swap'`<br />
+<Since v="3.6.0" />
+</p>
+
+Returns the fallback strategy for View Transitions.
+
+See the guide on [Fallback control](/en/guides/view-transitions/#fallback-control) for how to chose and configure the fallback behavior.
+
+## View Transition Events
+
+<p><Since v="3.6.0" /></p>
+
+### `astro:before-preparation` event
+
+An event dispatched at the beginning of a navigation using View Transitions. This event happens before any request is made and any browser state is changed.
+
+This event has the attributes:
+- [`info`](#info)
+- [`state`](#state-option)
+- [`sourceElement`](#sourceelement-option)
+- [`navigationType`](#navigationtype)
+- [`direction`](#direction)
+- [`from`](#from)
+- [`to`](#to)
+- [`formData`](#formdata)
+- [`loader`](#loader)
+
+Read more about how to use this event on the [View Transitions guide](/en/guides/view-transitions/#astrobefore-preparation).
+
+### `astro:after-preparation`
+
+An event dispatched after the next page in a navigation using View Transitions is loaded.
+
+This event has no attributes.
+
+Read more about how to use this event on the [View Transitions guide](/en/guides/view-transitions/#astroafter-preparation).
+
+### `astro:before-swap`
+
+An event dispatched after the next page is parsed and prepared and linked into a document in preparation for the transition but before any content is swapped between the documents.
+
+This event can't be canceled. Calling `preventDefault()` is a no-op.
+
+This event has the attributes:
+- [`info`](#info)
+- [`state`](#state-option)
+- [`sourceElement`](#sourceelement-option)
+- [`navigationType`](#navigationtype)
+- [`direction`](#direction)
+- [`from`](#from)
+- [`to`](#to)
+- [`viewTransition`](#viewtransition)
+- [`swap`](#swap)
+
+Read more about how to use this event on the [View Transitions guide](/en/guides/view-transitions/#astrobefore-swap).
+
+### `astro:after-swap`
+
+An event dispatched after the contents of the page have been swapped but before the view transition ends.
+
+The history entry and scroll position have already been updated when this event is triggered.
+
+### `astro:page-load`
+
+An event dispatched after a page completes loading, be it from a navigation using View Transition or native to the browser.
+
+When view transitions is enabled in the page, code that would normally execute on `DOMContentLoaded` should be changed to execute on this event.
```

Comment:
We're being a bit inconsistent by using both "view transitions" and "View Transitions", the latter appearing more often. I'd make all occurrences the same for consistency's sake. Also, a few nits.

```suggestion
An event dispatched after the contents of the page have been swapped but before the view transition ends.

The history entry and scroll position have already been updated when this event is triggered.

### `astro:page-load`

An event dispatched after a page completes loading, be it from a navigation using View Transitions or native to the browser.

When View Transitions is enabled on the page, code that would normally execute on `DOMContentLoaded` should be changed to execute on this event.
```

Reply:
view transitions always lower case UNLESS it's the proper name of the View Transitions API. (which we are probably renaming in 5.0 anyway...) I'll go through and do a sweep for this!